### PR TITLE
Remove requirement to have a verified email address before receiving email

### PR DESCRIPTION
**Pull request by jpaddison3**
The change in `renderEmail.tsx` is the only change that affects the EA Forum. The rest removes said functionality for LessWrong, to bring them into harmonization with the Forum. This is a somewhat extreme version of my general policy to, instead of hacking in `if (isEAForum)`s, to instead implement the desired behavior on both the Forum and LW, if LW is willing. In this case it's ambiguous whether LW will want this, but based on discussion in #ea_forum_joint, LW is open to it, at least.

I've kept in email verification for LW instead of removing it, in case it's helpful to not lose all that information. We have already removed it.

### CODE REVIEW

File: packages/lesswrong/lib/vulcan-users/helpers.ts [hunk 1]
```diff
@@ -70,5 +70,8 @@ export const userNumberOfItemsInPast24Hours = async function<T extends DbObject>
 };
 
 export const userFindByEmail = async function(email: string): Promise<DbUser|null> {
-  return await mongoFindOne("Users", { email: email });
+  const userByPlainEmail = await mongoFindOne("Users", { email: email });
+  if (userByPlainEmail) return userByPlainEmail;
+  const userByEmailsArray = await mongoFindOne("Users", { 'emails.address': email });
+  return userByEmailsArray;
```
Comment (jpaddison3) [inline packages/lesswrong/lib/vulcan-users/helpers.ts:76]: This is the function guarantees that there aren't duplicate email addresses, and is used on user signup. It's pretty important that we return users by their emails array or we'd get duplicates.

File: packages/lesswrong/server/emails/renderEmail.tsx [hunk 2]
```diff
@@ -309,8 +308,6 @@ export function reasonUserCantReceiveEmails(user: DbUser): string|null
 {
   if (!user.email)
     return "No email address";
-  if (!userEmailAddressIsVerified(user))
-    return "Address is not verified";
```
Comment (jpaddison3) [inline packages/lesswrong/server/emails/renderEmail.tsx:313]: By removing this reason for not sending emails, the EA Forum now will send emails to users without verified emails.

**Review by jcmustin — APPROVED** [review]
Seems good to me!

File: packages/lesswrong/lib/vulcan-users/helpers.ts [hunk 3]
```diff
@@ -70,5 +70,8 @@ export const userNumberOfItemsInPast24Hours = async function<T extends DbObject>
 };
 
 export const userFindByEmail = async function(email: string): Promise<DbUser|null> {
-  return await mongoFindOne("Users", { email: email });
+  const userByPlainEmail = await mongoFindOne("Users", { email: email });
+  if (userByPlainEmail) return userByPlainEmail;
+  const userByEmailsArray = await mongoFindOne("Users", { 'emails.address': email });
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/lib/vulcan-users/helpers.ts:75]: Oh yeah. It's pretty magic.

File: packages/lesswrong/lib/collections/users/custom_fields.ts [hunk 4]
```diff
@@ -702,7 +701,7 @@ addFieldsDict(Users, {
     type: Boolean,
     optional: true,
     group: formGroups.emails,
-    control: 'EmailConfirmationRequiredCheckbox',
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/lib/collections/users/custom_fields.ts:705]: Yes — thanks

**Pull request merged**